src/declaration.d

```diff
@@ -1016,6 +1016,8 @@ extern (C++) class VarDeclaration : Declaration
 {
     Initializer _init;
     uint offset;
+    uint sequenceNumber;            // order the variables are declared
+    __gshared uint nextSequenceNumber;   // the counter for sequenceNumber
```

Comment:
Also seems to work for nested scopes. Could we have some weird semantic order of nested functions or so that conflicts with this sequence number scheme?